src/test/scala/fr/davit/capturl/IriSpec.scala

```diff
@@ -153,4 +153,9 @@ class IriSpec extends FlatSpec with Matchers {
     Iri("path?query#", Lazy) shouldBe Iri("path?query#", Strict)
   }
 
+  "Iri" should "ignore leading and trailing whitespace" in {
+    Iri(" http://example.com/path\t", Lazy) shouldBe Iri("http://example.com/path")
+    Iri(" http://example.com/path\t", Strict) shouldBe Iri("http://example.com/path")
```

Comment:
A few more test cases with leading (or trailing respectively) whitespace only won't hurt.

Reply:
Wouldn't that essentially just test the `trim` method though?

Reply:
Using `trim` is implementation, the test does and sould not not know this. (A bad implementation could check whether the string starts with whitespace and only trim in this case, e.g.)